### ISSUE TITLE
Edit related objects

### ISSUE THREAD

**Issue by steff-o** [origo-map/origo#1315]
**Is your feature request related to a problem? Please describe.**
Sometimes data for an object is distributed over serveral tables in the database. When editing attributes, origo has no perception of when objects from different layers have a realtion in the database. It prevents a data model to have 1-to-many relations if they are to be edited in origo or forces the user to manually edit foreign keys attributes or use a flattened data model which requires a fixed maximum number of columns.

**Describe the solution you'd like**
In featureInfo related tables should be displayed as an attribute with an expandable tree to show the attributes from the related table. This would be recursive so that a related table can have related tables, and so on.

When editing, related tables should be editable from the parent's attribut editor by the means of an expanding tree or similar. It should also be possible to add and remove new instances in the child layer.

**Describe alternatives you've considered**
Flatten the datamodel. But that does not scale well for multi-level relations and child tables with many attributes. Also it would not look appealing.

**Additional context**
This feature would resemble the functionality in ArcMap which has the capability to follow relations. In the ESRI-world it does that from defined relations in the database. But since we're using WFS that has no perception of relations, this feature would require manually configuring origo with the foreign key relation.

**Comment by tonnyandersson** [comment]
Obviously, I'm not a db ninja, but isn't this possible to set up in the database, with related tables, triggers, editable views and stuff? @jokd, any suggestions?

**Comment by steff-o** [comment]
Well I'm not a db ninja either. More like a db sumo wrestler with bad technique, but messing around in the database only takes you that far. 

The main objective here is to create a dynamic GUI that adapts to the data and makes it easy to create applications to display and edit attributes in a complex data model, especially when there are _1-to-many-relationships_. It would also enable editing tables without a geometry if they are related to a feature that can be selected for editing.

Editable views does not exist on all platforms and generally only supports _1-to-1-relations_. 

Triggers can be used, but that requires editing a flattened "shadow" table that can use triggers to update the actual tables and it may require a complex synch to rebuild the flattened table when something changes in the actual tables. We have actually tried this, and it quickly became a mess and highly non-general.

Geoserver has a concept of complex features which supports defining complex relations, but they are only readonly, and there is still no support for displaying them i origo.

A light version of this feature would be to make it possible to build custom forms and inject edits to the edit handler. Still it would require some alterations to origo to support it as the only way to do that is to emit an event on the _window_-object, which we have established is not a good idea as it is not an official api. It could also require some support for custom edit forms and maybe some logic in edit store to deal with relations involving newly created objects if that can't be sorted out in the custom form.
